src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryMap.cs

```diff
@@ -0,0 +1,296 @@
+    /// 5) put all private methods at the end.
+    /// </summary>
+    public class SharedMemoryMap : IDisposable
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// <see cref="MemoryMappedFile"/> containg the header + content of this <see cref="SharedMemoryMap"/>.
+        /// </summary>
+        private readonly MemoryMappedFile _mmf;
+
+        /// <summary>
+        /// Name of the <see cref="MemoryMappedFile"/> backing this <see cref="SharedMemoryMap"/>.
+        /// </summary>
+        private readonly string _mapName;
+
+        private readonly IMemoryMappedFileAccessor _mapAccessor;
+
+        public SharedMemoryMap(ILogger logger, IMemoryMappedFileAccessor mapAccessor, string mapName, MemoryMappedFile mmf)
+        {
+            _logger = logger;
+            _mapName = mapName;
+            _mmf = mmf;
+            _mapAccessor = mapAccessor;
+        }
+
+        /// <summary>
+        /// Write content from the given <see cref="Stream"/> into this <see cref="SharedMemoryMap"/>.
+        /// The number of bytes of content being written must be less than or equal to the size of the <see cref="SharedMemoryMap"/>.
+        /// </summary>
+        /// <param name="content"><see cref="Stream"/> containing the content to write.</param>
+        /// <returns>Number of bytes of content written.</returns>
+        private async Task<long> PutStreamAsync(Stream content)
+        {
+            long contentLength = content.Length;
+            byte[] contentLengthBytes = BitConverter.GetBytes(contentLength);
+
+            try
+            {
+                using (MemoryMappedViewStream mmv = _mmf.CreateViewStream())
+                {
+                    await mmv.WriteAsync(contentLengthBytes, 0, SharedMemoryConstants.LengthNumBytes);
+                    await content.CopyToAsync(mmv, SharedMemoryConstants.MinBufferSize);
+                    await mmv.FlushAsync();
+                }
+
+                return contentLength;
+            }
+            catch
+            {
+                // This exception can be triggered by:
+                // 1) Size is greater than the total virtual memory.
+                // 2) Access is invalid for the memory-mapped file.
+                _mapAccessor.Delete(_mapName, _mmf);
+            }
+
+            _logger.LogError($"Cannot put bytes: {_mapName}");
+            return 0;
+        }
+
+        /// <summary>
+        /// Write content from the given <see cref="byte[]"/> into this <see cref="SharedMemoryMap"/>.
+        /// The number of bytes of content being written must be less than or equal to the size of the <see cref="SharedMemoryMap"/>.
+        /// </summary>
+        /// <param name="content"><see cref="byte[]"/> containing the content to write.</param>
+        /// <returns>Number of bytes of content written.</returns>
+        public async Task<long> PutBytesAsync(byte[] content)
+        {
```

Comment:
remove `async` and return `return PutStreamAsync(contentStream);` without await whereever possible